cloudify_ssh_key/operations.py

```diff
@@ -110,24 +127,25 @@ def create(**_):
     if store_public_key_material:
         ctx.instance.runtime_properties[PUBLIC_KEY_EXPORT] = \
             public_key_export
+
     if store_private_key_material:
         ctx.instance.runtime_properties[PRIVATE_KEY_EXPORT] = \
             private_key_export
 
 
 @operation(resumable=True)
 def delete(**_):
-
     # remove keys only if created on previous step
     key_name = ctx.instance.runtime_properties.get(SECRETS_KEY_NAME)
-    if key_name:
+    if key_name and ctx.instance.runtime_properties.get(SECRETS_KEY_OWNER):
         private_name = '{0}_private'.format(key_name)
         if _get_secret(private_name):
             _delete_secret(private_name)
         public_name = '{0}_public'.format(key_name)
         if _get_secret(public_name):
             _delete_secret(public_name)
         del ctx.instance.runtime_properties[SECRETS_KEY_NAME]
+        del ctx.instance.runtime_properties[SECRETS_KEY_OWNER]
 
```

Comment:
Let's add a message letting the user know that this operation is not being executed and why.